plugin/dca/config.go

```diff
@@ -0,0 +1,59 @@
+// plugin/dca/config.go
+package dca
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+)
+
+type PluginConfig struct {
+	Type    string `yaml:"type"`
+	Version string `yaml:"version"`
+	RpcURL  string `yaml:"rpc_url"`
+	Uniswap struct {
+		V2Router string `yaml:"v2_router"`
+		Deadline int64  `yaml:"deadline"`
+	} `yaml:"uniswap"`
+}
+
+func loadPluginConfig(basePath string) (*PluginConfig, error) {
+	// First try the base path from config
+	if basePath != "" {
+		configPath := filepath.Join(basePath, "dca.yml")
+		if data, err := os.ReadFile(configPath); err == nil {
+			return parseConfig(data)
+		}
+	}
+
+	// Fallback to default system path
+	configPath := filepath.Join("/etc/vultisig", "dca.yml")
+	data, err := os.ReadFile(configPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read config file: %w", err)
+	}
+	return parseConfig(data)
+}
+
```

Comment:
For config parsing and discover, we shouldn't drop down to regular yaml parsing

Please take a look at how viper is used in `config/config.go`.

For adding paths, use `viper.AddConfigPath`, and similarly use `SetConfigName` based on the input parameters

This kind of structure makes it very very difficult to deploy - using viper means we can override with ENV vars, merge files, and do all the various things you would typically do with a configuration flow without having to rewrite the code each time to support something